### DISCUSSION TITLE
How to display SVG?

### DISCUSSION THREAD

**Discussion by markusgritsch** [lona-web-org/lona#341] [Q&A] [answered]
When I run the following simple View,
```Python
from lona_picocss.html import HTML, H1, Button
from lona_picocss import install_picocss
from lona import App, View

app = App(__file__)


install_picocss(app, debug=True)  # remove 'debug' in production

@app.route('/')
class HelloWorldView(View):
    def handle_button_click(self, input_event):
        self.button.set_text('I was clicked')

    def handle_request(self, request):
        self.button = Button(
            'Click Me',
            handle_click=self.handle_button_click,
            _style={'width' : '8em'},
        )

        self.svg = HTML('''<svg width="400" height="100">
  <circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow" />
</svg>''')

        return HTML(
            H1('Hello World'),
            self.button,
            self.svg
        )


app.run(host='0.0.0.0')
```

 the SVG after the button is not displayed:

![grafik](https://user-images.githubusercontent.com/2684183/219975052-b634942f-6f71-4a18-b7b5-1fc78110481b.png)

However, if I save the webpage with the Browser to a static HTML file and load this again into the browser, it renderes fine:
![grafik](https://user-images.githubusercontent.com/2684183/219975143-38ec2e01-ab60-4492-b765-677ead5a9b83.png)

I cannot spot any difference in the Inspector.

When I use `interactive=False` in the route, it is rendered, but why is the SVG not rendered in the interactive View, and how can it be achieved?

**Comment by fscherf** [comment]
Hi @markusgritsch!

Currently, SVGs are not officially supported, using `lona.html`. What is the problem here? The client uses [Document.createElement()](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement), for any node that is about to be rendered. This includes the `svg` tag, and all tags like `circle` within. To make SVGs work, the client would have to use [Document.createElementNS()](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElementNS?retiredLocale=de) for the the svg tag.

The problem disappears on non-interactive views, because HTML from non-interactive views is handled as plain text by the browser.

I integrated svg icons for lona-picocss this weekend, but i used JavaScript to load the SVGs ( https://github.com/lona-web-org/lona-picocss/commit/f3530a4fe82f832270b97cebba5ec8c924bbe0fd#diff-7505eda20484b0ab66eb31727c7d72dad0fbf8ce824cbbbdafceffbb446c4221)

I think we should promote this discussion to an issue, and add the "enhancement" flag

**Comment by markusgritsch** [comment, marked as answer]
Discussion continues in issue https://github.com/lona-web-org/lona/issues/343.
